### ISSUE TITLE
Швидкодія: є варіанти прискорити?

### ISSUE THREAD

**Issue by a-vasyliev** [a-vasyliev/clock_img_maker#10]
Мої пропозиції:
1. Використовувати більше потужності компа.
2. Потоки, кілька дизайнів одночасно.

Збирає просто не те шоб ніч.. А кілька днів 5-7 фонів, 250 дизайнів. Якщо це не буде супер складно, я був би вдячний, якби трохи швидше працювало.

**Comment by mallhate** [comment]
є 2 варіанти:
1. розбити на процеси. це не супер складно, але потребує більше к-сть оперативної памяті.
2. перед роботою з дизайнами зменшити їх до, скажімо, 1200х1200 розміру. можливе погіршення якості годинника(але це не точно, можна зробити експеримент)

**Comment by a-vasyliev** [comment]
Зроби будь ласка розбивку на процеси. Буду дуже вдячний!

**Comment by mallhate** [comment]
спробую

**Comment by mallhate** [comment]
рекомендую ще подумати про зменшення розміру дизайну
зараз 3500х3500. вихідне зображення думаю максимум 800х800. тоїсть в 2 рази збільшити якість = 1600х1600. 3500х3500 = 12.25М пікселів, 1600х1600 = 2.56М. від цього можна виграти приблизно %50+ до швидкодії, а й можливо більше

**Comment by mallhate** [comment]
додав розбивку на процеси, але поки програму звичайним ctrl+c не знакрити

**Comment by mallhate** [comment]
на моїй машині кожен процес займає по 500мб в оперативі. надіюсь у вас не xeon на 18 ядер...

**Comment by a-vasyliev** [comment]
Found 38 design(s); 5 background(s)
max workers = 4
Traceback (most recent call last):
File "<�string>", line 1, in <�module>
File "multiprocessing\spawn.py", line 105, in spawn_main
File "multiprocessing\spawn.py", line 115, in _main
TypeError: new() missing 2 required positional arguments: 'lang' and 'tkey'
Traceback (most recent call last):
File "run.py", line 420, in <�module>
mm = MakerManager(func=worker_func, texture_image=texture_image, clock_face_image=clock_face_image)
File "run.py", line 224, in init
self.add_process()
File "run.py", line 229, in add_process
t.start()
File "multiprocessing\process.py", line 105, in start
File "multiprocessing\context.py", line 223, in _Popen
File "multiprocessing\context.py", line 322, in _Popen
File "multiprocessing\popen_spawn_win32.py", line 65, in init
File "multiprocessing\reduction.py", line 60, in dump
BrokenPipeError: [Errno 32] Broken pipe

--
Андрей

**Comment by a-vasyliev** [comment]
При цьому в Докері наче працює добре, на Вінді зламалось.

**Comment by mallhate** [comment]
А яка вінда? Чи достатньо оперативної памяті?

**Comment by mallhate** [comment]
Скільки в докері воркеків поставило?

**Comment by mallhate** [comment]
Не можу в себе повторити помилку

**Comment by a-vasyliev** [comment]
RAM 16 GB.

--
Андрей

**Comment by mallhate** [comment]
поставив 10ку на віртуалку. все працює(

**Comment by a-vasyliev** [comment]
Зібрав геть інший набір даних та налаштувань. Тільки запускаю start.bat відразу бачу помилку. Може ти запускаєш не через start.bat?

**Comment by a-vasyliev** [comment]
Може в репі не та версія Пітона? Може в Пітоні що в репо немає якогось модуля?

**Comment by a-vasyliev** [comment]
Встановив останню версію Пітона та Піллоу. Не спрацювало. Ось що знайшов: https://github.com/python-pillow/Pillow/issues/1434

**Comment by mallhate** [comment]
З бібліотекою pillow все добре. Дочірній процес не зміг запуститись. А через докер всі ядра загружаються на 100%? Можливо є смисл просто через докер на Вашому комп'ютері працювати?

**Comment by mallhate** [comment]
> Встановив останню версію Пітона та Піллоу. Не спрацювало. Ось що знайшов: [python-pillow/Pillow#1434](https://github.com/python-pillow/Pillow/issues/1434)
 Глобально встановили?

**Comment by a-vasyliev** [comment]
Так, глобально

**Comment by a-vasyliev** [comment]
І запустив глобально `python run.py`

**Comment by a-vasyliev** [comment]
Так, в результаті зараз в мене на ноуті (MacOS) та на ігровому компі (Win10) розгорнуто Docker. Проте підозрюю, що це сильно впливає на швидкість.

**Comment by a-vasyliev** [comment]
Отже, можу дати доступ до AnyDesk. Встанови собі і маякни, я скажу як доступитись до машини.
